spmd/tensor/dispatch.py

```diff
@@ -111,20 +111,12 @@ def _reshape_alias(
 ] = {torch.ops.aten._reshape_alias.default: _reshape_alias}
 
 
-def operator_dispatch(
+def prepare_inputs(
```

Comment:
Is the goal to use prepare_inputs directly, and bypass operator_dispatch, when using make_fx?

If this is the case, then we will probably miss some operators' implementations that directly use DTensor.

Could you provide a quick pseudo-code of the intended call sequence?

Reply:
This is how it is used in the PR on top

https://github.com/pytorch/tau/blob/a06ce4426bbddb84dc75eb9f0c10894c5c80bf41/test/spmd/test_tracing.py#L267-L301

we use `make_fx` to trace two things 1. redistributed inputs 2. local op. IIUC, we don't redistribute output at the moment in DT? If we do that in the future, we will also needs to add that to trace as well. 

Ideally, I wanted to trace `dispatch_with_local_tensors(local_args, arg_specs, output_specs)`, which will trigger redistribute on input, local comp op, redistribute on output.

Reply:
The issue i see is that if we do it this way we will be missing 1) the decompositions; 2) the custom op implementations (those that don't have a propagation rule, but that require direct DTensor-aware implementation). I think for some ops we will actually need those.

Would it be possible to work around make_fx limitations somehow and still be able to trace the implementation of DTensor at some level?